File: rb/features.rb
# MercadoBitcoin SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MercadoBitcoinFeatures
  def self.make_feature(name)
    case name
    when "base"
      MercadoBitcoinBaseFeature.new
    when "test"
      MercadoBitcoinTestFeature.new
    else
      MercadoBitcoinBaseFeature.new
    end
  end
end

File: py/mercadobitcoin_sdk/features.py
# MercadoBitcoin SDK feature factory

from mercadobitcoin_sdk.feature.base_feature import MercadoBitcoinBaseFeature
from mercadobitcoin_sdk.feature.test_feature import MercadoBitcoinTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MercadoBitcoinBaseFeature(),
        "test": lambda: MercadoBitcoinTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
